iModelJsNodeAddon/presentation/ECPresentationSerializer.cpp

```diff
@@ -1119,6 +1131,365 @@ rapidjson::Document IModelJsECPresentationSerializer::_AsJson(ContextR ctx, Node
+    rapidjson::Value idsJson;
+    idsJson.SetArray();
+    for (auto const& id : boundQueryIdSet.GetSet())
+        idsJson.PushBack(rapidjson::Value(id.ToString(BeInt64Id::UseHex::Yes).c_str(), json.GetAllocator()), json.GetAllocator());
+    json.AddMember("value", idsJson, json.GetAllocator());
+    return json;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+rapidjson::Document IModelJsBoundQueryValueSerializer::_ToJson(BoundECValueSet const& boundECValueSet, rapidjson::Document::AllocatorType* allocator) const
+    {
+    auto const& values = static_cast<ECValueVirtualSet const*>(boundECValueSet.GetSet().get())->GetValues();
+    rapidjson::Document json(allocator);
+    json.SetObject();
+    json.AddMember("type", "ValueSet", json.GetAllocator());
+    json.AddMember("valueType", rapidjson::Value(values.empty() ? 0 : GetECValueTypeName((*values.begin()).GetPrimitiveType()).c_str(), json.GetAllocator()), json.GetAllocator());
+    rapidjson::Value valuesJson;
+    valuesJson.SetArray();
+    for (auto const& value : values)
+        valuesJson.PushBack(GetJsonFromECValue(value, &json.GetAllocator()), json.GetAllocator());
+    json.AddMember("value", valuesJson, json.GetAllocator());
+    return json;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+rapidjson::Document IModelJsBoundQueryValueSerializer::_ToJson(BoundRapidJsonValueSet const& boundRapidJsonValueSet, rapidjson::Document::AllocatorType* allocator) const
+    {
+    auto const& set = static_cast<RapidJsonValueSet const&>(*boundRapidJsonValueSet.GetSet());
+    rapidjson::Document json(allocator);
+    json.SetObject();
+    json.AddMember("type", "ValueSet", json.GetAllocator());
+    json.AddMember("valueType", rapidjson::Value(GetECValueTypeName(set.GetValuesType()).c_str(), json.GetAllocator()), json.GetAllocator());
+    json.AddMember("value", rapidjson::Value(set.GetValuesJson(), json.GetAllocator()), json.GetAllocator());
+    return json;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+// @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+std::unique_ptr<BoundQueryValue> IModelJsBoundQueryValueSerializer::_FromJson(BeJsConst const &json)
+    {
+    if (!json.isObject() || !json.hasMember("type"))
+        return nullptr;
+
+    Utf8CP type = json["type"].asCString();
+    if (0 == strcmp("ECValue", type))
+        {
+        auto type = ParsePrimitiveType(json["valueType"].asCString());
+        if (type == 0)
+            return nullptr;
+        rapidjson::Document doc; // TODO: change to BeJsConst after converting RapidJson usage to BeJsConst

```

Comment:
this is repeated a lot. please extract into a method and comment why it's needed

Reply:
Done.